streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -471,6 +472,10 @@ private void replaceStreamThread(final Throwable throwable) {
 
     private void handleStreamsUncaughtException(final Throwable throwable,
                                                 final StreamsUncaughtExceptionHandler streamsUncaughtExceptionHandler) {
+        if (throwable instanceof NamedTopologyStreamsException) {
```

Comment:
we should consider exceptions that were not StreamExceptions as well I think

Reply:
Agreed